specification/vmware/Microsoft.AVS/models.tsp

```diff
@@ -0,0 +1,2364 @@
+enum MaintenanceProvisioningState {
+  ...ResourceProvisioningState,
+
+  @doc("is updating")
+  Updating,
+}
+
+@doc("type of the maintenance")
+enum MaintenanceType {
+  @doc("maintenance for vCenter Server Appliance")
+  VCSA,
+
+  @doc("maintenance for ESXi")
+  ESXI,
+
+  @doc("maintenance for NSX-T")
+  NSXT,
+}
+
+@doc("state of the maintenance")
+model MaintenanceState {
+  @doc("stage of the maintenance")
+  stage?: MaintenanceStage;
+
+  @doc("state of the maintenance stage")
+  stageStatus?: MaintenanceStageStatus;
+
+  @doc("Failure/Success info")
+  message?: string;
+
+  @doc("Time when current state started")
+  startedAt?: utcDateTime;
+
+  @doc("Time when current state ended")
+  endedAt?: utcDateTime;
+}
+
+@doc("stage of the maintenance")
+enum MaintenanceStage {
+  @doc("is creation")
+  Creation,
+
+  @doc("id customer approval")
+  CustomerApproval,
+
+  @doc("is pre-check")
+  PreCheck,
+
+  @doc("is execution")
+  Execution,
+
+  @doc("is reschedule")
+  Reschedule,
+
+  @doc("is cancellation")
+  Cancellation,
+}
+
+@doc("status of the maintenance")
+enum MaintenanceStageStatus {
+  @doc("is not started")
+  NotStarted,
+
+  @doc("is in progress")
+  InProgress,
+
+  @doc("is succeeded")
+  Succeeded,
+
+  @doc("is failed")
+  Failed,
+}
+
+@doc("Constraints for rescheduling maintenance")
+model MaintenanceRescheduleConstraints {
+  @doc("Deadline for upgrade")
+  @visibility("read")
+  deadline?: utcDateTime;
+
+  @doc("Freeze days")
+  @visibility("read")
+  freezeDays: plainDate[];
+}
+
+@doc("properties of a maintenance")
+model MaintenanceProperties {
+  @doc("type of maintenance")
+  @visibility("read")
+  component?: MaintenanceType;
+
+  @doc("Display name for maintenance")
+  @visibility("read")
+  displayName?: string;
+
+  @doc("Cluster ID for on which maintenance will be applied. Empty if maintenance is at private cloud level")
+  @visibility("read")
+  clusterId?: int32;
+
+  @doc("Link to maintenance info")
+  @visibility("read")
+  infoLink?: string;
+
+  @doc("Impact on the resource during maintenance period")
+  @visibility("read")
+  impact?: string;
+
+  @doc("The state of the maintenance")
+  @visibility("read")
+  state?: MaintenanceState;
+
+  @doc("Scheduled maintenance start time")
+  @visibility("read")
+  scheduledStartTime?: utcDateTime;
+
+  @doc("Estimated time maintenance will take in minutes")
+  @visibility("read")
+  estimatedDurationInMinutes?: int64;
+
+  @doc("The provisioning state")
+  @visibility("read")
+  provisioningState?: MaintenanceProvisioningState;
+
+  @doc("Constraints for rescheduling maintenance")
+  @visibility("read")
+  rescheduleConstraints?: MaintenanceRescheduleConstraints;
+}
+
+@doc("reschedule a maintenance")
+model MaintenanceReschedule {
+  @doc("reschedule time")
+  rescheduleTime?: utcDateTime;
+
+  @doc("rescheduling reason")
+  message?: string;
+}
+@doc("An iSCSI path resource")
+@parentResource(PrivateCloud)
+@singleton
```

Comment:
The addition of `@singleton` occurred after the most recent ARM approval.

Reply:
And that is okay to add because it always was, really, a singleton, is it?

Reply:
Yes, I believe it should be. I just missed it in the port to TypeSpec. It was introduced in Swagger format in https://github.com/Azure/azure-rest-api-specs-pr/pull/13757 . @balram2697, can you please confirm that this should be a `@singleton`?

Reply:
Yes IscsiPath is the singleton.